### PR TITLE
[WIP] Simplying RC Input Logic for Mapping PWM to Flight Modes

### PR DESCRIPTION
**Pull request by dlwalter**
**Describe problem solved by this pull request**
I had a problem that required an additional flight mode and found that the code to map the Flight Mode PWM channel to an actual Flight Mode was very convoluted.  It looks like it was designed to be very efficient and lightweight - but not easy to understand or expand on.

**Describe your solution**
Rather than a mapping equation that splits the 1000us PWM bandwidth into 160us chunks, I propose a series of if/else statements.  This could even let us manually define the pwm limits of each channel as a parameter.  This also simplifies adding additional flight modes.

**Describe possible alternatives**
The current implementation using the equation:

`manual.mode_slot = (((((_rc.channels[_parameters.rc_map_flightmode - 1] - slot_min) * num_slots) + slot_width_half) / (slot_max - slot_min)) + (1.0f / num_slots));
`

There is still a mapping that must occur from PWM space in us to -1,+1.  The if/else rc_input comparisons could be done in pwm space to simplify setting the mode ranges or implementing the limits as parameters.

### CODE REVIEW

File: src/modules/sensors/rc_update.cpp [hunk 1]
```diff
@@ -349,26 +349,41 @@ RCUpdate::rc_poll(const ParameterHandles &parameter_handles)
...
-						     (slot_max - slot_min)) + (1.0f / num_slots)) + 1;
-
-				if (manual.mode_slot > num_slots) {
-					manual.mode_slot = num_slots;
-				}
+				/* determine flight mode slot index from rc input */
+  				float rc_channel_val = _rc.channels[_parameters.rc_map_flightmode-1];
+
+  				/* each 0.1 represent 50us of PWM.  */
+  				if (rc_channel_val >= -1.05f && rc_channel_val < -0.68f) {  //PWM 1000 - 1160
```
Comment (dagar) [inline src/modules/sensors/rc_update.cpp]: I'm not sure about these magic numbers. Maybe we can find a line inbetween the heavily condensed convoluted equation unrolling it with magic numbers?

Comment (dlwalter) [reply src/modules/sensors/rc_update.cpp]: Yeah, exactly, I had the same idea as I was putting together the PR.  Just wanted some feedback if this is viable before I put more effort in.

I think if we just map _rc.channels into pwm space then the comparisons would be on us of PWM.  So the first if statement would be 

`if (rc_channel_val >= 1000f && rc_channel_val < 1160)`

Comment (dlwalter) [reply src/modules/sensors/rc_update.cpp]: @dagar let me know you opinion if this works better for determining the pwm bands.

**Comment by LorenzMeier** [comment]
Thanks for the contribution, but unfortunately I'm closing this as stale now. Please feel free to re-open a new PR if you feel strongly about it.

**Pull request closed without merging**